.github/workflows/sdk.yaml

```diff
@@ -46,27 +46,77 @@ jobs:
-        run: cargo clean && cargo test --package google-cloud-gax --no-default-features
-      - name: Test gax with all features
-        run: cargo clean && cargo test --package google-cloud-gax --all-features
-      - name: Test gax-internal with no features
-        run: cargo clean && cargo test --package google-cloud-gax-internal --no-default-features
-      - name: Test gax-internal with common features
-        run: cargo clean && cargo test --package google-cloud-gax-internal --features _internal_common
-      - name: Test gax-internal with http features
-        run: cargo clean && cargo test --package google-cloud-gax-internal --features _internal_http_client
-      - name: Test gax-internal with gRPC features
-        run: cargo clean && cargo test --package google-cloud-gax-internal --features _internal_grpc_client
-      - name: Test gax-internal with all features
-        run: cargo clean && cargo test --package google-cloud-gax-internal --all-features
       - name: Running tests for key crates
         run: cargo clean && cargo test
       - name: Build full workspace
         run: cargo clean && cargo build --workspace
       - name: Running tests for full workspace
         run: cargo clean && cargo test --workspace
         if: github.event_name == 'push'
+  features:
+    strategy:
+      matrix:
+        os: ['ubuntu-24.04']
+        rust-version: ['rust:current']
+    runs-on: ${{ matrix.os }}
+    env:
+      RUSTDOCFLAGS: "-D warnings"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo
+          key: ${{ github.job }}-${{ runner.os }}-${{ matrix.rust-version }}-cargo-${{ hashFiles('Cargo.lock', '.github/workflows/sdk.yaml') }}
+      - name: Setup Rust ${{ matrix.rust-version }}
+        run: |
+          set -e
+          rustup toolchain install ${{ fromJson(env.GHA_RUST_VERSIONS)[matrix.rust-version] }}
+          rustup default ${{ fromJson(env.GHA_RUST_VERSIONS)[matrix.rust-version] }}
+          rustup component add clippy
+      - run: rustup default ${{ fromJson(env.GHA_RUST_VERSIONS)[matrix.rust-version] }}
+      - name: Display Cargo version
+        run: cargo version
+      - name: Display rustc version
+        run: rustup show active-toolchain -v
+      - name: Test gax with features disabled
+        run: |
+          set -e
+          cargo clean
+          for sub in test doc; do
+            cargo "${sub}" --package google-cloud-wkt --no-default-features
+            cargo "${sub}" --package google-cloud-wkt --no-default-features --features prost
+            cargo "${sub}" --package google-cloud-wkt --no-default-features --features chrono
+            cargo "${sub}" --package google-cloud-wkt --no-default-features --features time
+            cargo "${sub}" --package google-cloud-wkt --all-features
+          done
+          cargo clippy --no-deps --package google-cloud-wkt --all-features -- --deny warnings
```

Comment:
Can `clippy` go inside the loop?

Reply:
Cannot think of a way to do so.  It needs the `-- --deny warnings` at the end, which the other `sub` commands do not accept.

It may be nicer if there was a way to get all the features of a package, but I cannot find an easy way to do so.